Automattic/jetpack
# On non-standard responses stop sending sync requests for 1 minute.

Currently we don't back-off sending  sync actions to WP.com based on the response. This update ties into the recently implemented Retry-After logic so that if we get a non 200 response to sync requests that we stop sending sync actions for 1 minutes. 

In the future we can get more granular if we need to based on if it was a 429, invalid markup, repeat failures, etc.

Note that we first check for the Retry-After header so this only will set a 1 minute delay if the response doesn't tell us differently.

#### Changes proposed in this Pull Request:
* Don't send sync requests for 1 minutes if a non-200 response is received.

#### Does this pull request change what data or activity we track or use?
* NO

#### Testing instructions:

* Sandbox your jetpack connection to your WP.com sandbox
* On your sandbox update the `sync_actions` function of `Jetpack_XMLRPC_Server` to return a non 200 response. Hint return_error.
* Update a post on your test site
* verify the jp_sync_retry_after_sync option is populated
* Update additional posts
* Verify no sync actions sent for the duration of the 1 minutes
* Remove the modification to the receive function.
* Verify that the sync actions flow after the option has expired and the option does not get set.
